### PR TITLE
[test] Fix flakes and speed up prebuild tests

### PR DESCRIPTION
**Pull request by WVerlaek**
## Description

Fixes flakiness in the prebuild e2e tests, where the prebuild was getting to a `stopped` phase too quickly before we were waiting for the workspace to stop.

Solved by adding a new `WaitForWorkspaceOpt` parameter that waits for the workspace to stop after launching it, so we always observe the last workspace (stopped) status even when the prebuild exits immediately.

This also lets us speed up the prebuild tests by removing the 10s wait for some prebuild tasks.

## Related Issue(s)

Relates to WKS-153 and WKS-227

## How to test

```
cd test
go test -v ./tests/components/ws-manager/...
```

I've ran this a number of times without flakes, where previously it was commonly failing

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

**Review by akosyakov — APPROVED** [review]
to unblock from IDE side, review from workspace is required still

/hold

**Review by aledbf — APPROVED** [review]

**Comment by kylos101** [comment]
Removing the hold, so that we can have the goodness.

**Pull request merged**
